### PR TITLE
Supporting Kubernetes 1.26

### LINKED ISSUE #665: Support for Kubernetes 1.26
The code-base must be aligned to support Kubernetes v1.26, also considering that starting from 1.25 version `v1beta1` of the `EndpointSlice` resource has been deprecated.

### PR DESCRIPTION
**Pull request by prometherion**
Closes #665.

### CODE REVIEW

**Review by maxgio92 — APPROVED** [review]
LGTM

Let's run it!

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        k8s-version: ['v1.16.15', 'v1.17.11', 'v1.18.8', 'v1.19.4', 'v1.20.7', 'v1.21.2', 'v1.22.4', 'v1.23.6', 'v1.24.1']
+        k8s-version: ['v1.16.15', 'v1.17.11', 'v1.18.8', 'v1.19.4', 'v1.20.7', 'v1.21.2', 'v1.22.4', 'v1.23.6', 'v1.24.7', 'v1.25.3', 'v1.26.0']
```
Comment (bsctl) [inline .github/workflows/e2e.yml:35]: do we need to run E2E tests for no more supported k8s releases? https://kubernetes.io/releases/patch-releases/#non-active-branch-history

**Review by bsctl — CHANGES_REQUESTED** [review]
please document supported releases https://github.com/clastix/kamaji/blob/master/docs/content/reference/versioning.md

File: .github/workflows/e2e.yml (hunk 1 above)
Comment (prometherion) [reply .github/workflows/e2e.yml:35]: It's effortless for us to maintain those old versions, and we should be proud of this!

**Comment by prometherion** [comment]
@bsctl noticed the wrong comment, please, approve this PR so I can merge it.

**Review by bsctl — APPROVED** [review]
LGTM

**Pull request merged**